test/karma/test-app/util.ts

```diff
@@ -141,52 +236,93 @@ export function setupDomTests(document: Document) {
           const elm = document.createElement('div');
           elm.innerHTML = this.responseText;
           frag.appendChild(elm);
-          app.innerHTML = elm.innerHTML;
+          testElement.innerHTML = elm.innerHTML;
 
-          const tmpScripts = app.querySelectorAll('script') as NodeListOf<HTMLScriptElement>;
-          for (let i = 0; i < tmpScripts.length; i++) {
-            const script = document.createElement('script') as HTMLScriptElement;
-            if (tmpScripts[i].src) {
-              script.src = tmpScripts[i].src;
-            }
-            if (tmpScripts[i].hasAttribute('nomodule')) {
-              script.setAttribute('nomodule', '');
-            }
-            if (tmpScripts[i].hasAttribute('type')) {
-              script.setAttribute('type', tmpScripts[i].getAttribute('type')!);
+          /**
+           * Re-generate script tags that are embedded in the loaded HTML file.
+           *
+           * Doing so allows JS files to be loaded (via script tags), when the HTML is served, without having to configure
+           * Karma to load the JS explicitly. This is done by adding the host/port combination to existing `src`
+           * attributes.
+           *
+           * Before:
+           * ```html
+           * <script type="module" src="/index.6127a5ed.js"></script>
+           * ```
+           *
+           * After:
+           * ```html
+           * <script src="http://localhost:9876/index.547a265b.js" type="module"></script>
+           * ```
+           */
+          const parseAndRebuildScriptTags = () => {
+            const tempScripts: NodeListOf<HTMLScriptElement> = testElement.querySelectorAll('script');
+            for (let i = 0; i < tempScripts.length; i++) {
```

Comment:
could this be `for (let tempScript of tempScripts)` too ?

Reply:
Unfortunately this can't be either - at least not as written. `querySelectorAll` returns a [`NodeList`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList), which isn't iterable.  We _could_ make a new array from it (a là `Array.from()`) - I didn't feel strongly either way and left it as is. Thoughts?